src/tpm2_pytss/utils.py

```diff
@@ -71,6 +71,80 @@ def make_credential(
     return (credblob, secret)
 
 
+def credential_to_tools(
+    id_object: Union[TPM2B_ID_OBJECT, bytes],
+    encrypted_secret: Union[TPM2B_ENCRYPTED_SECRET, bytes],
+) -> bytes:
+    """
+    Converts an encrypted credential and an encrypted secret to a format that TPM2-tools can handle.
+
+    The output can be used in the credential-blob parameter of the tpm2_activatecredential command.
+
+    Args:
+        id_object: The encrypted credential area.
+        encrypted_secret: The encrypted secret.
+
+    Returns:
+        A credential blob in byte form that can be used by TPM2-tools.
+    """
+    data = bytearray()
+
+    # Add the header, consisting of the magic and the version.
+    data.extend(int(0xBADCC0DE).to_bytes(4, "big") + int(1).to_bytes(4, "big"))
+
+    if isinstance(id_object, bytes):
+        id_object = TPM2B_ID_OBJECT(id_object)
+    if isinstance(encrypted_secret, bytes):
+        encrypted_secret = TPM2B_ENCRYPTED_SECRET(encrypted_secret)
+
+    # Add the id object and encrypted secret.
+    data.extend(id_object.marshal())
+    data.extend(encrypted_secret.marshal())
+
+    return data
+
+
+def tools_to_credential(
+    credential_blob: bytes,
+) -> Tuple[TPM2B_ID_OBJECT, TPM2B_ENCRYPTED_SECRET]:
+    """
+    Convert a TPM2-tools compatible credential blob.
+
+    Args:
+        credential_blob: A TPM2-tools compatible credential blob.
+
+    Returns:
+        A tuple of (TPM2B_ID_OBJECT, TPM2B_ENCRYPTED_SECRET)
+    """
+    magic = int.from_bytes(credential_blob[0:4], byteorder="big")
+    if magic != 0xBADCC0DE:
+        raise ValueError(f"bad magic, expected 0xBADCC0DE, got 0x{magic:X}")
+    version = int.from_bytes(credential_blob[4:8], byteorder="big")
+    if version != 1:
+        raise ValueError(f"bad version, expected 1, got {version}")
+
+    # Read the length of the id object.
+    id_object_len = int.from_bytes(credential_blob[8:10], byteorder="big")
+
+    # Read the id object including the length.
+    id_object_bytes = credential_blob[8 : 10 + id_object_len]
+
+    # Read the length of the encrypted secret.
+    encrypted_secret_len = int.from_bytes(
+        credential_blob[10 + id_object_len : 12 + id_object_len], byteorder="big"
+    )
+
+    # Read the encrypted secret including the length.
+    encrypted_secret_bytes = credential_blob[
+        10 + id_object_len : 12 + id_object_len + encrypted_secret_len
+    ]
+
+    id_object, _ = TPM2B_ID_OBJECT.unmarshal(id_object_bytes)
+    encrypted_secret, _ = TPM2B_ENCRYPTED_SECRET.unmarshal(encrypted_secret_bytes)
+
+    return id_object, encrypted_secret
```

Comment:
Skip the int.from_bytes here and use the offset from unmarshal, so something lile:
```python
offset = 8
id_object, off = TPM2B_ID_OBJECT.unmarshal(credential_blob[offset:])
offset += off
encrypted_secret = TPM2B_ENCRYPTED_SECRET.unmarshal(credential_blob[offset:])
```

Reply:
Done! Also amended it to the existing commit.